File: Day-Four/javascript/Credit-Card-Validator/CreditCardValidator.js
//funitons

function getCardType(cardNumber){

    let cardType = String(cardNumber)
    if (cardType.startsWith("4")){
        return "Visa Card";
    }
    else if (cardType.startsWith("5")){
        return "MasterCard";
    }
    else if(cardType.startsWith("37")){
        return "American Express Card";
    }
    else if(cardType.startsWith("6")){
        return "Discover Card";
    }
    return "Invalid card";
}

module.exports = {getCardType}

function getCardLength(cardNumber){

    let cardType = String(cardNumber);
    let count = 0;
    for(let letters of cardType){
        count+= 1;
    }
    return count;
}


module.exports = { getCardType, getCardLength };

function isValid(cardNumber){
    let cardType = String(cardNumber);
    let numbers = []
    let total = 0
    let digit = 0
    let sum = 0
    let new_sum = 0
    const size = cardType.length;

    for(let count = size - 2; count >= 0; count -= 2 ){
        numbers.push(Number(cardType[count]) * 2);
    }
    
    
    for(let number of numbers){
        if(number > 9){
            let firstDigit = Math.trunc(number / 10);
            let secondDigit = number % 10;
            total += firstDigit + secondDigit;
        }
        else{
            total += number
        }
    }

    for(let count = size - 1; count >= 0; count -= 2){
        
        let number = cardType[count]
        sum += Number(number)
    }
    new_sum = sum + total;

    if(new_sum % 10 ==  0){
        return "Valid";
    }
return "Invalid"
}

module.exports = { getCardType, getCardLength, isValid  };

File: Day-Four/python/Credit-Card-Validator/credit_card_validator.py

def get_card_type(card_number):

    card_type = str(card_number)

    if card_type.startswith("4"):
        return "Visa Card"
    elif card_type.startswith("5"):
        return "MasterCard"
    elif card_type.startswith("37"):
        return "American Express Card"
    elif card_type.startswith("6"):
        return "Discover Card"

    return "Invalid card"

def get_card_length(card_number):

    return len(str(card_number))

def get_card_validity(card_number):
    
    total = 0
    valid_total = 0
    odd_total = 0
    card_number = str(card_number)
    numbers = []
    size = len(str(card_number))
    for count in range(size - 2, -1, -2):

        numbers.append(int(card_number[count]) * 2)
        
    for number in numbers:
    
        number = str(number)
        if len(number) == 2:
                
            for digit in number:
                digit = int(digit)
                total += digit
        else:
            total += int(number)
            
    for index in range(size - 1, -1, -2):
    
        odd_total += int(card_number[index])

    valid_total = odd_total + total

    if valid_total % 10 == 0:
        return "Valid"

    return "Invalid"
    


